streams/src/test/java/org/apache/kafka/streams/integration/ProcessingExceptionHandlerIntegrationTest.java

```diff
@@ -0,0 +1,400 @@
+            final TestInputTopic<String, String> inputTopic = driver.createInputTopic("TOPIC_NAME", new StringSerializer(), new StringSerializer());
+            inputTopic.pipeKeyValueList(events, Instant.EPOCH, Duration.ZERO);
+
+            assertEquals(expected.size(), processor.theCapturedProcessor().processed().size());
+            assertIterableEquals(expected, processor.theCapturedProcessor().processed());
+
+            final MetricName dropTotal = droppedRecordsTotalMetric();
+            final MetricName dropRate = droppedRecordsRateMetric();
+
+            assertEquals(2.0, driver.metrics().get(dropTotal).metricValue());
+            assertTrue((Double) driver.metrics().get(dropRate).metricValue() > 0.0);
+        }
+    }
+
+    @Test
+    public void shouldContinueWhenProcessingExceptionOccursAtMiddle() {
+        final List<KeyValue<String, String>> events = Arrays.asList(
+            new KeyValue<>("ID123-2", "ID123-A2"),
+            new KeyValue<>("ID123-1", "ID123-A1"),
+            new KeyValue<>("ID123-3", "ID123-A3"),
+            new KeyValue<>("ID123-4", "ID123-A4")
+        );
+
+        final List<KeyValueTimestamp<String, String>> expected = Arrays.asList(
+            new KeyValueTimestamp<>("ID123-2", "ID123-A2", 0),
+            new KeyValueTimestamp<>("ID123-4", "ID123-A4", 0)
+        );
+
+        final MockProcessorSupplier<String, String, Void, Void> processor = new MockProcessorSupplier<>();
+        final StreamsBuilder builder = new StreamsBuilder();
+        builder
+            .stream("TOPIC_NAME", Consumed.with(Serdes.String(), Serdes.String()))
+            .map(KeyValue::new)
+            .mapValues(value -> value)
+            .process(runtimeErrorProcessorSupplierMock())
+            .process(processor);
+
+        final Properties properties = new Properties();
+        properties.put(StreamsConfig.PROCESSING_EXCEPTION_HANDLER_CLASS_CONFIG, ContinueProcessingExceptionHandlerMockTest.class);
+
+        try (final TopologyTestDriver driver = new TopologyTestDriver(builder.build(), properties, Instant.ofEpochMilli(0L))) {
+            final TestInputTopic<String, String> inputTopic = driver.createInputTopic("TOPIC_NAME", new StringSerializer(), new StringSerializer());
+            inputTopic.pipeKeyValueList(events, Instant.EPOCH, Duration.ZERO);
+
+            assertEquals(expected.size(), processor.theCapturedProcessor().processed().size());
+            assertIterableEquals(expected, processor.theCapturedProcessor().processed());
+
+            final MetricName dropTotal = droppedRecordsTotalMetric();
+            final MetricName dropRate = droppedRecordsRateMetric();
+
+            assertEquals(2.0, driver.metrics().get(dropTotal).metricValue());
+            assertTrue((Double) driver.metrics().get(dropRate).metricValue() > 0.0);
+        }
+    }
+
+    @Test
+    public void shouldContinueWhenProcessingExceptionOccursAtEnd() {
```

Comment:
What do you mean with "End" here?

Reply:
Not applicable anymore, tests have been changed